misc/formatter/modules/formatter-core/src/test/java/org/ballerinalang/formatter/core/expressions/StringTemplateExpressionsTest.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright (c) 2020, WSO2 Inc. (http://wso2.com) All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.ballerinalang.formatter.core.expressions;
+
+import org.ballerinalang.formatter.core.FormatterTest;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+import java.io.IOException;
+import java.nio.file.Paths;
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * Test the formatting of string template expressions.
+ *
+ * @since 2.0.0
+ */
+public class StringTemplateExpressionsTest extends FormatterTest {
```

Comment:
Why do we have a test class for each expression or construct type. Can we simply rely on a file structure to organize the test cases and run from a single test runner.

Reply:
Here the test classes extend a single test runner but there are test classes added for each subcategory since some of these subcategories contain a lot of test cases. But since this results in a large number of test classes, I will try to refactor this in the future.